forms/AddressFieldsetWithLookup/index.js

```diff
@@ -2,96 +2,116 @@
 
 export default React.createClass({
   displayName: 'AddressFieldsetWithLookup',
 
-  mixins: [i18nMixin],
+  mixins: [I18n],
 
   propTypes: {
     spacing: React.PropTypes.string,
-    address: React.PropTypes.object,
+    prefill: React.PropTypes.shape({
+      street_address: React.PropTypes.string,
+      extended_address: React.PropTypes.string,
+      locality: React.PropTypes.string,
+      region: React.PropTypes.string,
+      country_name: React.PropTypes.string,
+      postal_code: React.PropTypes.string
+    }),
     countryCode: React.PropTypes.string,
     prefix: React.PropTypes.string,
     onChange: React.PropTypes.func,
     onError: React.PropTypes.func,
-    validations: React.PropTypes.object
+    showError: React.PropTypes.bool,
+    storeLocally: React.PropTypes.bool,
+    validations: React.PropTypes.shape({
+      street_address: React.PropTypes.array,
+      extended_address: React.PropTypes.array,
+      locality: React.PropTypes.array,
+      region: React.PropTypes.array,
+      country_name: React.PropTypes.array,
+      postal_code: React.PropTypes.array
+    })
   },
 
-  getDefaultProps () {
+  getDefaultProps() {
     return {
       spacing: 'loose',
       prefix: '',
-      address: null,
+      prefill: null,
+      showError: false,
+      countryCode: 'au',
       onChange: () => {},
       onError: () => {},
       validations: {}
     }
   },
 
-  getInitialState () {
+  getInitialState() {
+    let countryCode = this.props.countryCode.toUpperCase()
     return {
-      countryCode: this.props.countryCode,
-      address: this.props.address
+      countryCode,
+      country: find(countries, country => country.value === countryCode),
+      address: this.props.prefill
     }
   },
 
-  getCountryName(value) {
-    return (find(countries, (country) => {
-      return country.value === value
-    }) || countries[0]).label
+  componentDidMount() {
+    let props = this.props
+    let errors = this.state.errors
+    props.onChange(props.prefill)
+    props.onError(includes(errors, true) || isEmpty(errors))
   },
 
-  clearAddress () {
-    this.setState({
-      address: null
-    })
+  clearAddress() {
+    this.setState({ address: null })
+    this.props.onChange(null)
+    this.props.onError(true)
   },
 
-  setEmptyAddress () {
-    let countryCode = this.refs.lookup.state.selectedCountry.value
+  setEmptyAddress() {
     this.setState({
-      countryCode,
       address: {
-        country_name: this.getCountryName(countryCode),
+        street_address: this.state.queryString,
+        country_name: this.state.country.label,
         paf_validated: false
       }
     })
   },
 
-  handleLookupChange (address) {
-    let countryCode = this.refs.lookup.state.selectedCountry.value
-    this.setState({
-      countryCode,
-      address
-    }, () => {
-      this.refs.fieldset.validate()
-      this.props.onChange(address)
-    })
+  handleError(err) {
+    if (typeof err !== 'boolean') {
```

Comment:
Maybe we should think about casting values before they make it into functions?
Just as a pattern for the future, pretend like this stuff is type checked... ?